### ISSUE TITLE
Automatic updating doesn't work for me (Sketch 49.2)

### ISSUE THREAD

**Issue by eliran-ran** [DWilliames/paddy-sketch-plugin#98]
#### Expected behavior:
I have a badge with 8px padding (Text + shape layer grouped together). The background layer is named "background [8]".
When changing the text, I would expect the background shape to automatically resize.
"Automatic updating" is turned on. 
I'm using Sketch 49.2

#### Actual behavior:
The background shape will resize only if I run the command "Enter padding for selection" again and again.
I wonder what happened to the amazing "Automatic updating" feature?

**Comment by najiglass** [comment]
Maybe the same issue for me, Paddy "stops" working sometimes.
Here's what the console says : 
Error in command main Script at path /Users/xxxxxxxxx/Library/Application Support/com.bohemiancoding.sketch3/Plugins/Paddy.sketchplugin/Contents/Sketch/main.js does not contain a handler function named: selectionChanged.

**Comment by highvoltaged** [comment]
Yeah, it stopped updating here too, only when you double click to edit a text field.
Other cases there is no way to get itto update, just re entering a number.

**Comment by Raghav789** [comment]
I think every time a sketch update happens plugin starts to misbehave, fresh installation worked for me 
1.un-install the plugin and quit sketch(not just close you need to quit it)
2.Reinstall the plugin

**Comment by DWilliames** [comment]
Yeah I think there's a weird issue within Sketch. Because there definitely is a handler called `selectionChanged` @najiglass. In the couple of cases I came across it, I just restarted Sketch and re-installed the plugin and it worked. It seems to be an issue with how Sketch caches the plugin I think. 🤔 No too sure. But I don't think there's much I can do.

@eliran-ran The 'automatic updating' feature should definitely still exist.

Either way – just an FYI, I'm in the process of completely re-writing Paddy; which hopefully should take care of a lot of these issues; plus in my first couple of benchmarks the new version seems to be around 30x faster 🔥 

Bare with me as I try and bring the new version up to parity, and I promise it will be heaps better.
Thanks for your patience all!

**Comment by najiglass** [comment]
Wow that’s incredible news. Thanks for doing this.
Hope you could manage nested symbols as well at some point so I could get rid of Auto-layout (which is so slow) :)

**Comment by trvsss** [comment]
Awesome news @DWilliames. Thanks for the hard work!

**Comment by eliran-ran** [comment]
Awesome news indeed @DWilliames. Thank you for you efforts!

**Comment by DWilliames** [comment]
@najiglass Thanks 😄 
I've begun focusing on nested symbols; and so far I have managed to get it working to some degree. I've got quite a few more kinks to work out. But it's looking promising 👌

**Comment by Raghav789** [comment]
@DWilliames it's amazing to hear that you are working on new plugin but please don't forget your core ideal. I've been working on this huge project for about six months and anima total ruined my spec export.

> No special data manipulation: This simply resizes and moves your layers to optimal positions – it doesn't turn your groups into a special 'Stack group' or something; so other people or programs such as Zeplin, can read the Sketch file perfectly fine, without the need for this plugin too.

**Comment by najiglass** [comment]
I’d say David is a genius and can do better :) In any case it might be worth having an enable/disable nested layers option if we think it might break exports.

Raghav, have you tried with the latest Anima update, they claim to have fixed exports.

**Comment by Raghav789** [comment]
@najiglass I've tried but instead of reversing or baking only their 'stacks' groups and collapses, they break all groups, removing shapes with no fill and only shadow(I've got a shadow override as a symbol), removing layer blending properties etc. it's very messy.

**Comment by DWilliames** [comment]
@Raghav789 @najiglass – Thanks for the input. I should definitely be able to do it in a way that 'Nested symbols' would be completely optional; so you could toggle it on/off; depending on how important it is to the user. So they can make the decision, given the trade-offs. 👍

**Comment by Raghav789** [comment]
@DWilliames I know I extremely simplifying here how about an option to flat/bake/the transformation?which just detaches the symbols till the point of transformation of the nested symbol, since I haven't encountered any such errors so far. then the user can revert back and continue.
